### PR TITLE
DOCK-2391: redirect 500 error to 415 for google token endpoint

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
This PR returns a 415 error when an invalid json is provided for `/auth/tokens/google`.

**Review Instructions**
After the 1.14.1 hotfix is deployed to staging, execute curl commands with invalid JSON

```
curl -X 'POST' \
  'http://staging.dockstore.org/api/auth/tokens/google' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer <redacted>' \
  -d '"string"'
```

```
curl -X 'POST' \
  'http://staging.dockstore.org/api/auth/tokens/google' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer <redacted>' \
  -d '"{"'
```

**Issue**
https://ucsc-cgl.atlassian.net/browse/DOCK-2391

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-webservice/src/main/resources/swagger.yaml [hunk 1]
```diff
@@ -1368,6 +1368,8 @@ paths:
       summary: "Allow satellizer to post a new Google token to Dockstore."
       description: "A post method is required by satellizer to send the Google token"
       operationId: "addGoogleToken"
+      consumes:
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/swagger.yaml:1371]: A little curious how this change was created

**Review by denis-yuen — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java [hunk 2]
```diff
@@ -429,7 +433,12 @@ private boolean getRegisterFromSatellizerObject(JsonObject satellizerObject) {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, notes = "A post method is required by satellizer to send the Google token", response = Token.class)
     public Token addGoogleToken(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth Optional<User> authUser, @ApiParam("code") String satellizerJson) {
         Gson gson = new Gson();
-        JsonElement element = gson.fromJson(satellizerJson, JsonElement.class);
+        JsonElement element;
+        try {
+            element = gson.fromJson(satellizerJson, JsonElement.class);
+        } catch (JsonSyntaxException ex) {
+            throw new CustomWebApplicationException(INVALID_JSON, HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java:440]: Suggest logging the exception.

**Review by coverbeck — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]
Should this be a hotfix?

**Comment by hyunnaye** [comment]
> Should this be a hotfix?

Oops, it was originally a hotfix but I accidentally merged the develop ugh. I'll fix it

**Comment by hyunnaye** [comment]
Will delete this PR and and create a new one to cherry pick. New PR: https://github.com/dockstore/dockstore/pull/5573

**Pull request closed without merging**